TYPO3.Neos/Resources/Public/JavaScript/Content/Navigate/AbstractNodeTree.js

```diff
@@ -569,24 +569,29 @@ define(
 					parentNode = activeNode;
 				}
 
-				allowedNodeTypes = this._getAllowedChildNodeTypesForNode(parentNode);
+				allowedChildNodeTypes = this._getAllowedChildNodeTypesForNode(parentNode);
 
 				// Only show node types which inherit from the base node type(s).
 				// If the base node type is prefixed with "!", it is seen as negated.
+				var allowedNodeTypes = [];
+				var disallowedNodeTypes = [];
 				this.get('baseNodeType').split(',').forEach(function(nodeTypeFilter) {
 					nodeTypeFilter = nodeTypeFilter.trim();
-
-					allowedNodeTypes = allowedNodeTypes.filter(function (nodeTypeName) {
```

Comment:
To explain, before it looped over each node type from the baseNodeType filter meaning if you had two non-negate filter, it would remove everything. This also fixes it for the document node tree